### ISSUE TITLE
Breaks with apache2 v3.2.2

### ISSUE THREAD

**Issue by e100** [zuazo/postfixadmin-cookbook#10]
In apache2 3.2.2 apache.listen_ports and apache.listen_addresses are depreciated and this cookbook fails to run.
https://github.com/sous-chefs/apache2/blob/master/CHANGELOG.md

Quick Hack in apache2 recipe almost fixes the problem:
```
 def update_listen_ports(port)                              
-  return if node['apache']['listen_ports'].include?(port)  
-  node.set['apache']['listen_ports'] =                     
-    node['apache']['listen_ports'] + [port]                
+  return if node['apache']['listen'].include?( "*:#{port}")
+  node.default['apache']['listen'] =                       
+    node['apache']['listen'] + ["*:#{port}"]               
 end                                                        

```

I also had to manually edit the node to delete the listen_ports previously set by postfixadmin cookbook.

**Comment by zuazo** [comment]
Hi @e100 :wink:

I have to ask: On what platform? What's the exact error?

I'm testing it with `apache2` cookbook version `3.2.2` and seem to be working properly. See for example in travis build [204584045](https://travis-ci.org/zuazo/postfixadmin-cookbook/jobs/204584045) (ran today):

    836    [2017-02-23T14:14:10+00:00] INFO: Loading cookbooks [
                  netstat@0.1.0,
                  postfixadmin_test@0.1.0,
                  postfixadmin@3.0.0
                  [...]
                  apache2@3.2.2
                  [...]
              ]

**Comment by e100** [comment]
Ubuntu 14.04

I think the root cause might be a bug in the apache 3.2.2 cookbook.
If the node has apache2.listen_ports defined but not apache2.listen_address an exception is thrown from apache2 cookbook when it is executed.

Since this cookbook does a node.set on apache2.listen_ports it can trigger this problem.
The only way I could get things working on my node after upgrading to apache2 3.2.2 was the code change above combined with editing the node to delete the previously set listen_ports attribute.

**Comment by e100** [comment]
I also just realized that I am using older version of postfixadmin from supermarket so maybe the version on github is OK. Any chance you can update the supermarket?

**Comment by e100** [comment]
I think the latest version here on github is OK, I had issues when using 2.1.0 so this can be closed as invalid.

**Comment by zuazo** [comment]
The version currently in master will be a major bump that includes some breaking changes: update to PostfixAdmin 3, require Chef 12 Custom Resources, update some cookbook dependencies, etc.

I think I will need like a couple of weeks before releasing it, but I'm working hard on it.

As always, your feedback is really appreciated, @e100! Thanks! :smiley:

**Comment by zuazo** [comment]
Fix released in [`3.0.0`](https://github.com/zuazo/postfixadmin-cookbook/releases/tag/3.0.0).
